Prepare 9.3.3

bourtemb opened the pull request:
Update cppTango to release 9.3.3
Update TangoAccessControl to release 2.14
Update assets/README to show changes between 9.2.5a and 9.3.3
Update assets/TANGO_CHANGES for 9.3.3 with a minimal text explanation

bourtemb committed:
Update cpptango to release 9.3.3

bourtemb committed:
Update assets/README...
... to show changes between 9.2.5a and 9.3.3

bourtemb committed:
Update assets/TANGO_CHANGES for 9.3.3
Minimal text explanation in this version.

bourtemb committed:
Update TangoAccessControl to Release 2.14

Ingvord commented:
Should we merge #16 first?

Ingvord commented:
I would change rest-server to 1.9 as well

bourtemb commented:
Good question... I missed that one...

bourtemb committed:
Update rest-server to 1.9

Ingvord committed:
Update README

Ingvord commented:
There is a bunch of changes in the README. Maybe we should extract them to assets/CHANGES_LOG file or merge it with TANGO_CHANGES?

bourtemb commented:
> There is a bunch of changes in the README. Maybe we should extract them to assets/CHANGES_LOG file or merge it with TANGO_CHANGES?

I think there is a big work to be done to harmonize all these files and to make them up to date.

Some sections in the README file are overlapping what is/should be in other files.
I think all the installation instructions should be in the INSTALL file (which could be upgraded to INSTALL.md).
All the Changes should be in a separate file too.
It looks like there are different kinds of changes which were listed before: the release versions of the different SW parts and the bugs and changes in C++ Tango library.
There was not much information about the bugs and changes in other SW (Astor, JTango,...).

The questions are:
Do we change that?
How do we organize that?
What kind of information do we want to provide to the user?

I think there is a big work to be done to ensure the content of these files is up to date.

I guess it should not be too complicated to generate automatically a file from assets/distribution.properties which would be listing all the different versions used for the different provided Tango SW components.

About the Changelog, I am not sure about the level of details we would provide here...
If we provide some details, do we do it for the changes in all SW components, not only cppTango?

I guess it would be good to provide RELEASE NOTES too.
Again, we need to agree on the level of details we provide.

Ingvord commented:
From my experience in such aggregated distribution the best is to refer to every particular repo aka README, CHANGE_LOG, RELEASE_NOTES etc individually via a link. 

While here to provide an info on migration e.g. what needs to be done to ensure 9.2.5a -> 9.3.3 migration (possibly not much), but if have such an info for each Tango release it is much more convenient for users to track what needs to be done to migrate from Tango 7.1.2 (arbitrary number) to 9.3.3

I agree a discussion would be helpful

bourtemb committed:
Update to latest available release versions
Update Jive to          Release 7.21
Update ATK to           Release 9.3.2
Update AtkPanel to      Release 5.8
Update Astor to         Release 7.2.4
Update JSSHTerminal to  Release 1.13
Update Pogo to          Release 9.6.18
Update TangoDatabase to Release 5.9
Update tango_admin to   Release 1.14

bourtemb committed:
Downgrade Astor temporarily to release 7.2.3

bourtemb committed:
Fix missing separator error in common_target.opt

Fix the following error (a tab was missing):
common_target.opt:166: *** missing separator.

bourtemb committed:
Update build.xml
Use copy ant task instead of deprecated copydir task

bourtemb committed:
Cleanup POGO (#20)

Remove obsolete POGO templates
Remove pogo-6

bourtemb committed:
Update astor to release 7.2.5

mliszcz commented:
Hi @bourtemb, @Ingvord 

CC @piogor

What is the current status of 9.3.3? Have any critical issues been found?

There were no issues reported in kernel in last month (at least there's nothing new on github).
In source distribution we have #24 and #25 - root cause is known and possible corrections were proposed.
What about #21, #22 and #23 - do you plan these enhancements for 9.3.3?

We plan to include Tango 9.3.3 in Tangobox 9.3. I'm just wondering whether we could use this branch (`prepare-9.3.3`) and patch #24/#25 manually or should we wait for an official release.

bourtemb commented:
Hi @mliszcz,

I'm just coming back from holidays. I didn't have time yet to comment the PRs or issues you created.

No big issue were reported indeed.
I think there is still quite a lot of work to be done on the Tango distribution to release it.
We should definitely integrate #24 and #25.
I think we should also solve #21 and #23 at least.
#22 would be great too because it would avoid to generate a liblog4tango library which should no longer exist.
There is a big work to be done to update and cleanup files like the assets/README assets/TANGO_CHANGES and assets/INSTALL files.

Ingvord commented:
Hi, @mliszcz , @bourtemb ,

Here is my 50 cents:

#24, #25, #21 - must be included in 9.3.3 before release.

#23 I am not sure, maybe if it won't break anything should be added as well.

#22 will require considerable amount of manpower (as per my experience with #8) so probably will be done in the next release,  as well as other "big work" on assets.

mliszcz commented:
Hi @bourtemb, @Ingvord 

Thank you guys for your inputs. I'll submit a PR for #24 and #25 still today.
If no one will take care for #21, I may try to take a look next week - depends whether we will have spare cycles at S2Innovation.

PS. Reynald, I hope you had a great holiday!

mliszcz committed:
rename atkpanel to ATKPanel in java applications (fixes #25) (#27)

mliszcz committed:
Pass correct arguments to TangoRestServer (fixes #24) (#26)

bourtemb committed:
Remove zmq.hpp (#23) (#28)

Ingvord committed:
Upgrade rest-server to 1.11

mliszcz committed:
Add logback configuration file (#21) (#29)

bourtemb commented on distribution.properties:
```diff
@@ -22,7 +22,7 @@ dbbench-ver=1.3
 log-viewer-ver=2.0.5
 pogo-ver=9.6.18
-rest-server-ver=1.9
+rest-server-ver=1.11
```
@Ingvord , could you please also update the README?
https://github.com/tango-controls/TangoSourceDistribution/blob/prepare-9.3.3/assets/README

Ingvord replied:
Done in f409a5a

mliszcz committed:
Merge liblog4tango into libtango (change to convenience lib) (#30) (#31)

Change log4tango into a libtool convenience library.
liblog4tango.so is no longer generated, but log4tango objects are still compiled and are now linked into libtango.so

Ingvord committed:
Update README

mliszcz committed:
Remove my.cnf config file (tango-controls/TangoDatabase#16) (#35)

mliszcz committed:
Remove log4tango from tango.pc & makefiles; add dummy liblog4tango.so (#33)

picca commented:
Hello could you re-add the source of the documentation, in order to be able to re-generate the documention during the build process on the Debain bbuild farm.

thanks

picca commented:
Is it possible to provide a migration script of the database from n-1 to n, instead of  updating all migrations scripts at each release, thanks

Frederic

picca commented:
Here a faillure during the build

https://salsa.debian.org/science-team/tango/-/jobs/185931

picca commented:
```
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../lib/cpp/client -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/server -I/usr/include -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wl,-z,relro -I/usr/include  -D_TANGO_LIB -g -O2 -fdebug-prefix-map=/tmp/building/package=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o zmqeventsupplier.lo zmqeventsupplier.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../lib/cpp/client -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/server -I/usr/include -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wl,-z,relro -I/usr/include -D_TANGO_LIB -g -O2 -fdebug-prefix-map=/tmp/building/package=. -fstack-protector-strong -Wformat -Werror=format-security -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c zmqeventsupplier.cpp  -fPIC -DPIC -o .libs/zmqeventsupplier.o
zmqeventsupplier.cpp: In member function 'void Tango::ZmqEventSupplier::create_mcast_socket(std::__cxx11::string&, int, Tango::ZmqEventSupplier::McastSocketPub&)':
zmqeventsupplier.cpp:689:18: error: cannot convert 'zmq::socket_t' to 'void*'
     if (zmq_bind(*(ms.pub_socket),ms.endpoint.c_str()) != 0)
                  ^~~~~~~~~~~~~~~~
In file included from /usr/include/zmq.hpp:38,
                 from ../../../lib/cpp/client/event.h:41,
                 from ./tango.h:146,
                 from zmqeventsupplier.cpp:33:
/usr/include/zmq.h:482:32: note:   initializing argument 1 of 'int zmq_bind(void*, const char*)'
 ZMQ_EXPORT int zmq_bind (void *s_, const char *addr_);
                          ~~~~~~^~
make[6]: *** [Makefile:785: zmqeventsupplier.lo] Error 1
```

Ingvord commented:
@picca , thanks for the info and the efforts to build Debian package out of this branch. This is already addressed in [cppTango-530](https://github.com/tango-controls/cppTango/pull/530). I guess we will have to wait for a merge...

bourtemb commented:
> Is it possible to provide a migration script of the database from n-1 to n, instead of updating all migrations scripts at each release, thanks

I think we should still provide scripts for users migrating from Tango 7 and Tango 8.
But it's true that these scripts could also call the other scripts...
For instance, if a user wants to upgrade from Tango 7, the Tango 7 migration script would contain the DB queries to update from Tango 7 to Tango 8, and then simply call the Tango 8 migration script.
This would avoid code duplication.
Still, this means that we would need to update all the scripts at each source distribution release implying a DB change.

Why do you consider the update of all these scripts as a problem?

picca commented:
Hello,

this is juste that dbconfig-common does an incremental upgrade from on version to the other.
So I just need a script whcih does the migration from 9.2.5 to 9.3.3

DELETE FROM property_class WHERE class='DServer' AND count >= 11;
INSERT INTO property_class VALUES('DServer','AllowedAccessCmd',11,'EventConfirmSubscription',NOW(),NOW(),NULL);


Is it the only change  ?

bourtemb commented:
> Is it the only change ?

Yes, it is.

bourtemb commented:
Actually, I think there is also something related to the rest server:

```
#
# Create entry for TangoRestServer device server in device table
#

DELETE FROM device WHERE server='TangoRestServer/rest';

INSERT INTO device VALUES ('sys/rest/0',NULL,'sys','rest','0',0,'nada','nada','TangoRestServer/rest',0,'TangoRestServer','nada',NULL,NULL,'nada');
INSERT INTO device VALUES ('dserver/TangoRestServer/rest',NULL,'dserver','TangoRestServer','rest',0,'nada','nada','TangoRestServer/rest',0,'DServer','nada',NULL,NULL,'nada');
```

But maybe this is not yet in the update scripts.

bourtemb committed:
Add CMake opt files (#34) (#36)

* Add CMake opt files (#34)
Add cmake_common_target.opt and cmake_tango.opt to the distribution
in order to be able to generate CMakeLists.txt files for Device
Servers from Pogo.

* remove -std option from cmake_tango.opt

* Compile with -std=c++0x when using g++ < 6.1

bourtemb committed:
Update to latest available release versions (#37)

Update to latest available release versions
Update Jive to          Release 7.22
Update ATK to           Release 9.3.6
Update Pogo to          Release 9.6.23
Update rest-server to   Release 1.14
Update TangoDatabase to Release 5.11
Update JTango to 9.5.14
